src/parser/Parser.cpp

```diff
@@ -1084,6 +1103,15 @@ void Parser::resync(std::set<TokenType> types) {
     }
 }
 
+// Check expected next token
+void Parser::expect(std::set<TokenType> exp) {
+    auto type = scanner_.peek()->type();
+    if (exp.find(type) == exp.end()) {
+        auto token = scanner_.peek();
+        logger_.error(token->start(), "unexpected section "+ to_string(token->type()) + ".");
```

Comment:
Error message should be `unexpected token` in order to be consistent how parsing errors are reported elsewhere.